lib/devise_two_factor/strategies/two_factor_authenticatable.rb

```diff
@@ -21,6 +21,10 @@ def authenticate!
 
       def validate_otp(resource)
         return true unless resource.otp_required_for_login
+
+        # if fingerprints exists fallback to database strategy
+        return true if resource.browser_fingerprint[params['device_fp']].present?
```

Comment:
this line is not checking validation time span because outdated fingerpirint are removed in BrowserFingerprint::VAalidate